### PR TITLE
One time battles (rival battles)

### PR DESCRIPTION
**Pull request by Jaaslet**
This PR will add One time battles as town buttons. It can later be added as icons on the map, by someone else.
It also moves the ???-battle in Aether Paradise, from a gym to a one time battle
This PR does not cover Ultima's wish that the pokemons should changed, based on requirements (like the starter). I am planning to do a small rework, at some point, which changes how the champion is coded, so all gym leaders, champions and one time battles, can change pokemons, based on different requirements.

### CODE REVIEW

**Comment by RedSparr0w** [comment]
Haven't had the chance to fully look through everything yet,
Just a couple issues I noticed though:
- The location doesn't show as orange, which makes it hard to know where you need to go next in terms of progression:
image
- The `GO` Text offsets the Pokémon image:
image

File: src/components/temporaryBattleView.html [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            </div>
+        </h2>
+
+        <div id="temporaryBattleGoContainer">
+          <div class="row" style="display: inline-block; vertical-align: middle;">
+            <div class="col-sm-6 offset-sm-3">
+              <img style="width:100%" id="temporaryBattleGo" src="assets/gifs/go.gif">
+            </div>
+          </div>
+        </div>
```
Comment (RedSparr0w) [inline src/components/temporaryBattleView.html:39]: Could we also have this tied to the `Show gym GO` setting?

File: src/scripts/temporaryBattle/TemporaryBattleList.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+const temporaryBattleList: { [battleName: string]: TemporaryBattle } = {};
```
Comment (RedSparr0w) [inline src/scripts/temporaryBattle/TemporaryBattleList.ts:1]: Can we capitalize this as it's a global variable.
```suggestion
const TemporaryBattleList: { [battleName: string]: TemporaryBattle } = {};
```

File: src/modules/requirements/TemporaryBattleRequirement.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import Requirement from './Requirement';
+import * as GameConstants from '../GameConstants';
+
+export default class TemporaryBattleRequirement extends Requirement {
+    public battleName: string;
+    constructor(battleName: any, defeatsRequired = 1) {
+        super(defeatsRequired, GameConstants.AchievementOption.more);
+        this.battleName = battleName;
```
Comment (RedSparr0w) [inline src/modules/requirements/TemporaryBattleRequirement.ts:8]:
```suggestion
    constructor(public battleName: string, defeatsRequired = 1) {
        super(defeatsRequired, GameConstants.AchievementOption.more);
```

File: src/modules/GameConstants.ts [hunk 4]
```diff
@@ -21,6 +21,8 @@ export const MIN_LOAD_TIME = 0.5 * SECOND;
 export const MAX_LOAD_TIME = 20 * SECOND;
 export const MUTATION_TICK = 1 * SECOND;
 export const WANDER_TICK = 1.5 * SECOND;
+export const ONETIMEBATTLE_TIME = 60 * SECOND;
+export const ONETIMEBATTLE_TICK = 0.1 * SECOND;
```
Comment (RedSparr0w) [inline src/modules/GameConstants.ts:25]:
```suggestion
export const TEMP_TYPE_BATTLE_TIME = 60 * SECOND;
export const TEMP_TYPE_BATTLE_TICK = 0.1 * SECOND;
```
```suggestion
export const TEMP_BATTLE_TIME = 60 * SECOND;
export const TEMP_BATTLE_TICK = 0.1 * SECOND;
```

**Comment by Jaaslet** [comment]
I think i have fixed all your comments now

**Review by RedSparr0w — APPROVED** [review]
Looks good to me, thanks for the PR! 👍

**Pull request merged**